### PR TITLE
Update bdk-ffi to 0.5 (with TxBuilder APIs)

### PR DESCRIPTION
**Pull request by notmandatory**
### Description

Update bdk-ffi to https://github.com/bitcoindevkit/bdk-ffi/pull/123 and add new TxBuilder tests.

### Notes to the reviewers

I also fixed tests to use network and addresses for TESTNET to match BlockchainConfig. 

The tests must be a bit hacky until we have a way to use REGTEST for our Kotlin tests.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [x] I've added tests for the new feature
* [x] I've added docs for the new feature
* [ ] I've updated `CHANGELOG.md`

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
Looking at this right now. Should the submodule point to a tag on bdk-ffi? The `0.5.0` tag is [8a556d0ba0d5cd499b39dd65ca073229a45ffce2](https://github.com/bitcoindevkit/bdk-ffi/releases/tag/v0.5.0). Just not sure what our "official" procedure is or should be.

File: jvm/src/test/kotlin/org/bitcoindevkit/JvmLibTest.kt [hunk 1]
```diff
@@ -101,18 +101,53 @@ class JvmLibTest {
...
+            val faucetAddress = "tb1ql7w62elx9ucw4pj5lgw4l028hmuw80sndtntxt"
+            val txBuilder = TxBuilder().addRecipient(faucetAddress, 1000u).feeRate(1.2f)
+            val psbt = txBuilder.build(wallet)
+            wallet.sign(psbt)
+            val txid = wallet.broadcast(psbt)
+            println("https://mempool.space/testnet/tx/$txid")
+            assertNotNull(txid)
+        } else {
+            val depositAddress = wallet.getLastUnusedAddress()
+            println("Send more testnet coins to: $depositAddress")
```
Comment (thunderbiscuit) [inline jvm/src/test/kotlin/org/bitcoindevkit/JvmLibTest.kt:139]: When I first ran the tests the wallet was empty, but for some reason the `println()` statement doesn't get printed in the `else` branch in the same way the others in the `if` branch do get printed (if the wallet has funds, the print statements _do_ work).

So it wasn't clear to me why the test was failing at first, until I checked the balance manually. One way to fix this is by putting the message inside the `fail()` method, which worked for me:
```sh
JvmLibTest > walletTxBuilderBroadcast FAILED
    java.lang.AssertionError: Send more testnet coins to: tb1qvkc0mfqfpxhpyewzvkxdrn4fd8afse7frucs3h
        at org.junit.Assert.fail(Assert.java:89)
        at org.bitcoindevkit.JvmLibTest.walletTxBuilderBroadcast(JvmLibTest.kt:141)
```

**Comment by artfuldev** [comment]
@thunderbiscuit I was gonna pick that up and complete upgrading to 0.5 but happy to take help.

**Review by thunderbiscuit — CHANGES_REQUESTED** [review]
I agree using testnet for the tests is not ideal for now (the wallet will drain rather quickly if we always run tests), but I do think it's good enough for now. I suggested one small change to the error message for the `walletTxBuilderBroadcast` test. Otherwise everything worked well on my end.

**Comment by artfuldev** [comment]
I'm yet to upgrade to 0.5 tag which will add 2 new APIs for testing.

**Comment by notmandatory** [comment]
Until we can get a CI regtest setup (see #48) I think the best approach is if you're pushing changes, first run them locally and pre-fund the two testnet wallets for the `walletTxBuilderBroadcast` and `walletTxBuilderDrainWallet` tests.

File: jvm/src/test/kotlin/org/bitcoindevkit/JvmLibTest.kt (hunk 1 above)
Comment (notmandatory) [reply jvm/src/test/kotlin/org/bitcoindevkit/JvmLibTest.kt]: I moved the depositAddress message into the `fail()`.

**Comment by notmandatory** [comment]
After discussing with @thunderbiscuit commented out the `walletTxBuilderDrainWallet` test for now, it can be commented out and ran locally until we have a regtest CI setup working.

**Comment by artfuldev** [comment]
Do we need to add documentation?

**Review by thunderbiscuit — APPROVED** [review]
Tested ACK [a67b88a](https://github.com/bitcoindevkit/bdk-kotlin/pull/40/commits/a67b88a3ea99f2f136e441e1add937a9820eef9e). @artfuldev was there anything else you were wanting to add? Yesterday I think you were saying you wanted to do a few more things. I don't want my ACK to mean "merge" if you're not ready.

**Comment by notmandatory** [comment]
Rebased. Working on updating the `docs.patch` then will merge.

**Pull request merged**